google-cloud-spanner/src/main/resources/com/google/cloud/spanner/connection/PG_ClientSideStatements.json

```diff
@@ -0,0 +1,413 @@
+      "regex": "(?is)\\A\\s*(?:abort)(?:\\s+batch)\\s*\\z",
+      "method": "statementAbortBatch",
+      "exampleStatements": ["abort batch"],
+      "examplePrerequisiteStatements": ["start batch ddl"]
+    },
+    {
+      "name": "SET AUTOCOMMIT = TRUE|FALSE",
+      "executorName": "ClientSideStatementSetExecutor",
+      "resultType": "NO_RESULT",
+      "regex": "(?is)\\A\\s*set\\s+autocommit\\s*(?:=)\\s*(.*)\\z",
+      "method": "statementSetAutocommit",
+      "exampleStatements": ["set autocommit = true", "set autocommit = false"],
+      "setStatement": {
+        "propertyName": "AUTOCOMMIT",
+        "separator": "=",
+        "allowedValues": "(TRUE|FALSE)",
+        "converterName": "ClientSideStatementValueConverters$BooleanConverter"
+      }
+    },
+    {
+      "name": "SET SPANNER.READONLY = TRUE|FALSE",
+      "executorName": "ClientSideStatementSetExecutor",
+      "resultType": "NO_RESULT",
+      "regex": "(?is)\\A\\s*set\\s+spanner\\.readonly\\s*(?:=)\\s*(.*)\\z",
+      "method": "statementSetReadOnly",
+      "exampleStatements": ["set spanner.readonly = true", "set spanner.readonly = false"],
+      "setStatement": {
+        "propertyName": "SPANNER.READONLY",
+        "separator": "=",
+        "allowedValues": "(TRUE|FALSE)",
+        "converterName": "ClientSideStatementValueConverters$BooleanConverter"
+      }
+    },
+    {
+      "name": "SET SPANNER.RETRY_ABORTS_INTERNALLY = TRUE|FALSE",
+      "executorName": "ClientSideStatementSetExecutor",
+      "resultType": "NO_RESULT",
+      "regex": "(?is)\\A\\s*set\\s+spanner\\.retry_aborts_internally\\s*(?:=)\\s*(.*)\\z",
+      "method": "statementSetRetryAbortsInternally",
+      "exampleStatements": ["set spanner.retry_aborts_internally = true", "set spanner.retry_aborts_internally = false"],
+      "examplePrerequisiteStatements": ["set spanner.readonly = false", "set autocommit = false"],
+      "setStatement": {
+        "propertyName": "SPANNER.RETRY_ABORTS_INTERNALLY",
+        "separator": "=",
+        "allowedValues": "(TRUE|FALSE)",
+        "converterName": "ClientSideStatementValueConverters$BooleanConverter"
+      }
+    },
+    {
+      "name": "SET SPANNER.AUTOCOMMIT_DML_MODE = 'PARTITIONED_NON_ATOMIC'|'TRANSACTIONAL'",
+      "executorName": "ClientSideStatementSetExecutor",
+      "resultType": "NO_RESULT",
+      "regex": "(?is)\\A\\s*set\\s+spanner\\.autocommit_dml_mode\\s*(?:=)\\s*(.*)\\z",
+      "method": "statementSetAutocommitDmlMode",
+      "exampleStatements": ["set spanner.autocommit_dml_mode='PARTITIONED_NON_ATOMIC'", "set spanner.autocommit_dml_mode='TRANSACTIONAL'"],
+      "setStatement": {
+        "propertyName": "SPANNER.AUTOCOMMIT_DML_MODE",
+        "separator": "=",
+        "allowedValues": "'(PARTITIONED_NON_ATOMIC|TRANSACTIONAL)'",
+        "converterName": "ClientSideStatementValueConverters$AutocommitDmlModeConverter"
+      }
+    },
+    {
+      "name": "SET STATEMENT_TIMEOUT = '<duration>'|NULL",
```

Comment:
We were thinking of supporting the following:

```
SET STATEMENT_TIMEOUT = { '<INT8>{ s | ms | us | ns }' | INT8 | DEFAULT }
```

Do you think we can add the `DEFAULT` value / `INT8` (millis) value / remove `null`?

Reply:
Done.